src/dialects/ibmi/query-generator.js

```diff
@@ -130,21 +130,26 @@ export class IBMiQueryGenerator extends AbstractQueryGenerator {
   }
 
   dropTableQuery(tableName, options) {
-    let table = tableName;
+
     let schema;
 
-    if (typeof table === 'object') {
-      schema = table.schema || undefined;
-      table = table.table;
-    } else if (options.schema) {
-      schema = options.schema;
-    }
 
-    return `DROP TABLE IF EXISTS ${schema ? `"${schema}".` : ''}"${table}"`;
+     if (typeof tableName === 'object') {
+      schema = tableName.schema || options.schema || undefined;
```

Comment:
no other dialect accepts specifying the schema as part of the option bag, it's always part of `tableName`. You can drop `options` here.

(Not an issue for `describeTableQuery`, but we'll eventually need to drop the second parameter there too to be consistent)

Reply:
I've gone ahead and fixed this in #15235